src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -770,3 +770,191 @@ export async function getRefundRequests(
 
   return refundRequests;
 }
+
+/**
+ * @todo Improve type safety.
+ * @description Deserializes a serialized `UBAClientState` object.
+ * @param serializedUBAClientState Serialized `UBAClientState` object that for example gets returned
+ * when calling `updateUBAClient` and serializing the result.
+ * @returns Deserialized `UBAClientState` object. Specifically with `{ type: "BigNumber", value: "0x0" }`
+ * converted to `BigNumber` instances. And a correct `UBAFeeConfig` instance.
+ */
+export function deserializeUBAClientState(serializedUBAClientState: object): UBAClientState {
```

Comment:
I am not too happy with this function... Especially the types. I will try to refine them more but wanted to have something testable end2end soon. That's why I used so many `any`s here to move quickly.